src/librustc_driver/lib.rs

```diff
@@ -218,8 +218,7 @@ fn make_output(matches: &getopts::Matches) -> (Option<PathBuf>, Option<PathBuf>)
 
 // Extract input (string or file and optional path) from matches.
 fn make_input(free_matches: &[String]) -> Option<(Input, Option<PathBuf>)> {
-    if free_matches.len() == 1 {
-        let ifile = &free_matches[0][..];
+    if let Some(ifile) = free_matches.first() {
```

Comment:
Are you sure that `free_matches.len()` is always 1 or 0? This seems to change the behavior if it's, say, 2 or 3.